Common/Algorithm/Framework/Alphas/Analysis/AlphaAnalysisContext.cs

```diff
@@ -168,5 +173,25 @@ public override string ToString()
         {
             return $"{Alpha.Id}: {Alpha.GeneratedTimeUtc}/{Alpha.CloseTimeUtc} -- {Alpha.Score}";
         }
+
+        /// <summary>Serves as the default hash function. </summary>
+        /// <returns>A hash code for the current object.</returns>
+        /// <filterpriority>2</filterpriority>
+        public override int GetHashCode()
+        {
+            return Id.GetHashCode();
+        }
+
+        /// <summary>Determines whether the specified object is equal to the current object.</summary>
+        /// <returns>true if the specified object  is equal to the current object; otherwise, false.</returns>
+        /// <param name="obj">The object to compare with the current object. </param>
+        /// <filterpriority>2</filterpriority>
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) return false;
+            if (ReferenceEquals(this, obj)) return true;
+            if (obj.GetType() != GetType()) return false;
+            return Id.Equals(((AlphaAnalysisContext)obj).Id);
```

Comment:
Curious what is the main reason for this?

Reply:
`Equals` and `GetHashCode` are used in hashset and concurrent dictionary implementations for the key types. It's a performance boost to implement these members, otherwise they would all have the same hashcode and the hashset or concurrenct dictionary is then about as useful as a list